cli/pkg/kube/client.go

```diff
@@ -197,7 +200,53 @@ func (c *Client) DeleteOldOdigosSystemObjects(ctx context.Context, resourceAndNa
 	labelSelector := k8slabels.NewSelector().Add(*systemObject).Add(*notLatestVersion).String()
 	resource := resourceAndNamespace.Resource
 	ns := resourceAndNamespace.Namespace
-	return c.Dynamic.Resource(resource).Namespace(ns).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
-		LabelSelector: labelSelector,
-	})
+	cfg := ctrl.GetConfigOrDie()
+	versionSupported, err := isVersionSupported(cfg, 1, 23)
```

Comment:
`v1.23.0` should be put in a variable of type version with a name that suggests what feature is supported from that version